src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -6057,8 +6057,8 @@ static void Main()
             var x2Ref = GetReference(tree, "x2");
             Assert.Equal("string", model.GetTypeInfo(x2Ref).Type.ToDisplayString());
 
-            VerifyModelForDeconstructionLocal(model, x1, x1Ref);
-            VerifyModelForDeconstructionLocal(model, x2, x2Ref);
+            VerifyModelForDeconstruction(model, x1, LocalDeclarationKind.OutVariable, x1Ref);
+            VerifyModelForDeconstruction(model, x2, LocalDeclarationKind.OutVariable, x2Ref);
```

Comment:
Clearly, neither x1 nor x2 are out variables. This contradicts the proposed design and decreases its value even more since the kind now can be inaccurate. It feels like we need another special kind to cover scenarios like this in order to satisfy the proposed design.